### ISSUE TITLE
JEDEC Manufacturer's ID codes addition for unknown RAM manufacturers

### ISSUE THREAD

**Issue by x86fr** [memtest86plus/memtest86plus#22]
JEDEC JEP106 Manufacturer ID list is very long and many brands are defunct since years or even decades. 
We had to create a short list with the most well-known manufacturers. Others will reported as "Unknown".

If your RAM Module are detected as Unknown, or if you are a RAM manufacturer with a JEDEC ID Code and want Memtest86+ to detect your brand, please post a screen shot of the unknown module or your JEDEC ID here. It will added as soon as possible.

**Comment by Fantu** [comment]
on my test of today don't recognizes one ram module, also dmidecode don't show manufacturer field (show "0000"), is one module added after buying the notebook, is: https://www.amazon.it/gp/product/B0145WDNI4/
I don't know if is possible do something

**Comment by x86fr** [comment]
That's not uncommon for unbranded memory modules to have the manufacturer field not set (set to 0x0000 or 0x1f7f). 
I will add a special case to distinguish "Unbranded" from "Unknown".

**Comment by x86fr** [comment]
PS: can you post a screenshot?

**Comment by x86fr** [comment]
This has been solved some commit ago. It should not display anything when jedec_id is blank.

**Comment by sstscrypto** [comment]
We are a manufacturer of computer memory, but your software does not show our JEDEC ID, our JEDEC ID is 08d0

**Comment by x86fr** [comment]
Will be added on next release

**Comment by sstscrypto** [comment]
thank you so much

**Comment by sstscrypto** [comment]
> Will be added on next release

i download the Memtest86+ v6.00 Beta 3
no my JEDEC ID show in the test screen
My JEDEC ID is 08d0
Can we add our JEDEC ID to the list of recognizable ones?

**Comment by Nable80** [comment]
Is it actually 0x0850 ("Heoriady") which becomes 0x08D0 after adding a parity bit?
https://github.com/memtest86plus/memtest86plus/blob/main/system/jedec_id.h#L1101

**Comment by sstscrypto** [comment]
> Is it actually 0x0850 ("Heoriady") which becomes 0x08D0 after adding a parity bit? https://github.com/memtest86plus/memtest86plus/blob/main/system/jedec_id.h#L1101

Dear Heoriady:
In response to your request I have assigned the following JEDEC Manufacturer ID number to your 
company:
80 decimal (bank nine)
1101 0000 binary
D0 hex

These are my responses from JEDEC!

**Comment by sstscrypto** [comment]
[https://imgur.com/a/LVng66b](url)

**Comment by sstscrypto** [comment]
[screenshot](https://imgur.com/a/LVng66b)

**Comment by sstscrypto** [comment]
Can you delete the comment next time? in JEDEC.H

**Comment by x86fr** [comment]
Will do, but what's the name of your company ? Is "Heoriady" correct ?

**Comment by Nable80** [comment]
@sstscrypto I have an off-topic question regarding memory manufacturing, can I contact you somehow, please?

I can also ask it here if it's allowed by maintainers. Maybe someone else knows the answer.

**Comment by 01e3** [comment]
Can we also add 0x0313, "Golden Empire"?

![image](https://user-images.githubusercontent.com/105136494/180495334-ac3c857a-a10e-419e-b332-33671611326a.png)

If it helps, I can send a patch.

**Comment by x86fr** [comment]
> I can also ask it here if it's allowed by maintainers. Maybe someone else knows the answer.

Create a new discussion. I can also help (I have a Jedec ID for my company)

**Comment by Nable80** [comment]
> Create a new discussion. I can also help (I have a Jedec ID for my company)

Thank you very much! Created it there: https://github.com/memtest86plus/memtest86plus/discussions/127

**Comment by sstscrypto** [comment]
> @sstscrypto I have an off-topic question regarding memory manufacturing, can I contact you somehow, please?
> 
> I can also ask it here if it's allowed by maintainers. Maybe someone else knows the answer.

Sure you can contact me by E-mail hynix.cn@live.cn

**Comment by virxkane** [comment]
Is it possible to uncomment this line?
https://github.com/memtest86plus/memtest86plus/blob/18f12116c09d749944002548a1de635331f31054/system/jedec_id.h#L1327

![IMG_0285_cr](https://user-images.githubusercontent.com/36960933/188810549-09bf6964-45cd-48c4-b78c-2595d1180bb7.JPG)

**Comment by x86fr** [comment]
will be done on next update

**Comment by tatokis** [comment]
I have this [Mushkin DDR2 991760](https://www.poweredbymushkin.com/Home/index.php/catalog/memory/silverline/item/138-ddr2/144-991760) RAM. The vendor ID (0x577) is commented out and thus shows as unknown.

https://github.com/memtest86plus/memtest86plus/blob/c41159084d8e82fec018e1ff469e88c882eeacdb/system/jedec_id.h#L766

It appears that Avant Technology is still the correct vendor, as according to http://www.avanttechnology.com/Home/index.php/our-company:

> Avant, Edge, Mushkin and Avant de Mexico operate under the Avant Technology Corporate Umbrella.

In addition, memtest86+ v5 recognised the sticks as `F6456U61E5800F`. In v6 the field is completely blank (even after uncommenting the jedec id). I'm not sure which is the correct behaviour, since decode-dimms also shows no part number.

![IMG_20220914_041404](https://user-images.githubusercontent.com/6003656/190042501-97f6e4cd-13cb-4d92-9e40-5deec9bdf1ed.jpg)

I have attached the SPD in case it helps.
[991760.bin.gz](https://github.com/memtest86plus/memtest86plus/files/9581844/991760.bin.gz)

**Comment by x86fr** [comment]
I will add Avant Tech. for the next update. 

About the Part Number String not displaying, your SPD dump shows a malformed string. The two first bytes of the JEDEC Part Number String (Bytes #73-#90) are end-of-string characters. Your module's SPD is not JEDEC compliant and can't be decoded properly. Memtest86+ v5 used another way to display this string, without checking the end-of-string chars.

**Comment by 01e3** [comment]
So, what we have there:
 00 00 46 36 34 35 36 55 36 31 45 35 38 30 30  46 00 00

Even if we change the logic, I'm not convinced that F6456U61E5800F would be actually a desired thing to show. This does not look like anything that can be considered a part number,  even https://www.google.com/search?q=F6456U61E5800F says "About 0 results (0.21 seconds) ".

**Comment by CMCanavessi** [comment]
I don't know if it's the same module, but that part number is actually on the sticker, with an "AV" prefix and a "9-SPHP" and the end.

![image](https://user-images.githubusercontent.com/34553661/192316892-e1f7aae4-7527-475b-bf44-800a3deea866.png)

![image](https://user-images.githubusercontent.com/34553661/192316932-d81f6433-603d-4f24-89b4-50179429690b.png)

**Comment by 01e3** [comment]
Great catch!

http://www.avanttechnology.com/pdf/DataSheets/AVF6456U61E5800F9-AP.pdf says "AVF6456U61E5800F9"

So, the first two 00 00 characters should be "A" and "V" (0x41 and 0x56) - AV=AVANT, and perhaps we are missing "9" (0x39) at the end - REV=9.

Either EEPROM is corrupted or (more likely) coding / programming error?

**Comment by 01e3** [comment]
(just noticed your edit)

There seems to be multiple flavors of this module:
 - DDR2-800 CL5: http://www.avanttechnology.com/pdf/DataSheets/AVF6456U61E5800F9-AP.pdf
 - DDR2-800 CL6: http://www.avanttechnology.com/pdf/DataSheets/AVF6456U61E6800F9-AP.pdf

See also: http://www.avanttechnology.com/pdf/DDR2-Decoder.pdf

So, not a typo! :)

**Comment by CMCanavessi** [comment]
Nice, that clears it up a bit

**Comment by FranciscoPombal** [comment]
Is there a technical limitation that prevents adding the full known list? What limitations are in play here?

**Comment by x86fr** [comment]
95% of the list are defunct manufacturers and adding them would add hundreds of KB of useless strings to the code.

**Comment by FranciscoPombal** [comment]
Sorry, let me rephrase the question: what would be the performance/compatibility/etc implications of adding those KB's? What are the current limitations preventing such an addition from being "zero cost" (i.e. regressions are not observed elsewhere when those added strings are not actually being used/accessed in some way).

**Comment by daiaji** [comment]
> 95% of the list are defunct manufacturers and adding them would add hundreds of KB of useless strings to the code.

Not surprisingly, quite a few memory manufacturers (some of them even produce DRAM chips) were destroyed by dumping by Samsung after South Korea increased its subsidies to Samsung.
Qimonda and Elpida, these are the two more famous Samsung victims.
There are countless unknown DRAM manufacturers.
Let alone OEM.
To be honest, these JEDEC IDs may be the last traces they left in this world.: (

**Comment by Nable80** [comment]
Can you uncomment ID 0x0356, please? https://github.com/memtest86plus/memtest86plus/blob/6799bfb/system/jedec_id.h#L481

One of my memory modules has this identifier: ``Slot 0: 256MB DDR-400 - Unknown (0x356)``

**Comment by x86fr** [comment]
Will do before next update.

**Comment by BoDrate** [comment]
Could you uncomment 0x0671, "InnoDisk Corporation", please? We are using their memory modules.

**Comment by CMCanavessi** [comment]
> 95% of the list are defunct manufacturers and adding them would add hundreds of KB of useless strings to the code.

I was thinking about this. Would it be feasable to make that list of old manufacturers external? Say, memtest will look for a text file named a certain way, and if it exists, it will use it (together with its internal list). This way you could keep memtest with the more up-to-date list of vendors, but have the possibility of manually adding the older ones in a text file without altering the main binary. Best of both worlds imho.

**Comment by bjlockie** [comment]
![20230526_173946-memtest](https://github.com/memtest86plus/memtest86plus/assets/303873/d4fae83c-a132-4cd9-ab1b-173e39d4c39a)

It is Silicon Power DDR4 8GB 3200MHz (PC4-25600) CL22 SODIMM 260-Pin Gaming Laptop RAM Notebook Computer Memory.

**Comment by nickos222** [comment]
Hello guys, please uncomment the following IDs:
https://github.com/memtest86plus/memtest86plus/blob/18f12116c09d749944002548a1de635331f31054/system/jedec_id.h#L34
https://github.com/memtest86plus/memtest86plus/blob/18f12116c09d749944002548a1de635331f31054/system/jedec_id.h#L114
https://github.com/memtest86plus/memtest86plus/blob/18f12116c09d749944002548a1de635331f31054/system/jedec_id.h#L208

These have been verified on the real h/w with a patched memtest build.

**Comment by TAbdiukov** [comment]
I can confirm that the line for "**Strontium**", https://github.com/memtest86plus/memtest86plus/blob/18f12116c09d749944002548a1de635331f31054/system/jedec_id.h#L901

Needs to be uncommented, too.

* Memtest86+ nondetection, https://imgur.com/NqL7aGq

More Proof that this is Strontium (A Singaporean small flash memory enterprise):
* CPU-Z: https://imgur.com/mfr8xUB
* AIDA64: https://imgur.com/3l9zjQu

**Comment by dstarr5d** [comment]
![20231218_181802](https://github.com/memtest86plus/memtest86plus/assets/51419372/1069ca0e-eb2b-4014-b36e-fdf7ec3e7696)

This is four sticks of Teamgroup ‎CTCED532G6400HC32ADC01

**Comment by ed806** [comment]
Please add our Jedec ID: 0E51 brand: Xiaoli.AI
Thanks!

**Comment by BoDrate** [comment]
I am out of the office until 01/15/2024.

I will respond to your message when I return. Thank you & have a nice day!


Note: This is an automated response to your message  "Re:
[memtest86plus/memtest86plus] JEDEC Manufacturer's ID codes addition for
unknown RAM manufacturers (Issue #22)" sent on 12/27/2023 8:04:55 AM.

This is the only notification you will receive while this person is away.

**Comment by x86fr** [comment]
Add previous manufacturers added in commit 2d3b14ed1a5c75d080ff81547d6800b9f24847d3

**Comment by BoDrate** [comment]
I am out of the office until 01/15/2024.

I will respond to your message when I return. Thank you & have a nice day!


Note: This is an automated response to your message  "Re:
[memtest86plus/memtest86plus] JEDEC Manufacturer's ID codes addition for
unknown RAM manufacturers (Issue #22)" sent on 01/07/2024 3:23:16 PM.

This is the only notification you will receive while this person is away.

**Comment by grkngkc** [comment]
Please add our Jedec ID: 0E58h brand: Trium Elektronik Bilgi Islem San Ve Dis Tic Ltd.

**Comment by KevinKLL** [comment]
Please add our JEDEC Manufacturer's ID: 8FB3    Company: Xllbyte
in Memtest show:0xf33
![mem](https://github.com/memtest86plus/memtest86plus/assets/166833214/dca248dc-cd39-4cad-8b10-4f0c3f367c44)

Thanks!

**Comment by SSTC2024** [comment]
Hello @x86fr  
Could you please add our Jedec Manufacture Code into database ?
Our code: 8F37 
Name:  SSTC Technology and Distribution Inc   
Thank you very much

**Comment by nickos222** [comment]
Hello @x86fr
Please add the ID 0x1636 into the DB.
It is used on Lenovo DDR4 SODIMMs that are manufactured from various vendors.
Lenovo really likes to mix up memory vendors on notebooks that belong to the exact same model.
The memory module on the first PC is made from Samsung while the one on the second PC is made from SK-Hynix.

Thank you

**Comment by BoDrate** [comment]
I am out of the office until 01/13/2025.

I will respond to your message when I return.
Technical questions can be directed at ***@***.***
Thank you very much & have a nice day!


Note: This is an automated response to your message  "Re:
[memtest86plus/memtest86plus] JEDEC Manufacturer's ID codes addition for
unknown RAM manufacturers (Issue #22)" sent on 12/28/2024 12:11:27 AM.

This is the only notification you will receive while this person is away.

**Comment by Firminator** [comment]
Hi there, could add Thermaltake (US vendor)?
Created a commit @ https://github.com/memtest86plus/memtest86plus/pull/475 a few weeks ago.

**Comment by x86fr** [comment]
> Hello [@x86fr](https://github.com/x86fr) Please add the ID 0x1636 into the DB. It is used on Lenovo DDR4 SODIMMs that are manufactured from various vendors. Lenovo really likes to mix up memory vendors on notebooks that belong to the exact same model. The memory module on the first PC is made from Samsung while the one on the second PC is made from SK-Hynix.
> 
> Thank you

This is not a valid JEDEC Manufacturer ID according to JEP106 list. Lenovo ID is 0x0848. Please post a screenshot of Memtest86+ running on one of these module to grab the correct ID.

**Comment by KevinKLL** [comment]
Hello @x86fr Please add the JEDEC Manufacturer's ID: 8FB3 Company: Xllbyte
in Memtest show:0xf33
thank you very much!

**Comment by x86fr** [comment]
NTS: uncomment  { 0x0132, "Mushkin Enhanced Memory" },

**Comment by edenlzd** [comment]
Please add our JEDEC Manufacturer's ID: 0x875 Company: JUHOR
Thank you

![Image](https://github.com/user-attachments/assets/0a7b9e53-f8bb-4c16-9612-05c25c622df5)

**Comment by sarrchri** [comment]
> > Hello [@x86fr](https://github.com/x86fr) Please add the ID 0x1636 into the DB. It is used on Lenovo DDR4 SODIMMs that are manufactured from various vendors. Lenovo really likes to mix up memory vendors on notebooks that belong to the exact same model. The memory module on the first PC is made from Samsung while the one on the second PC is made from SK-Hynix.
> > Thank you
> 
> This is not a valid JEDEC Manufacturer ID according to JEP106 list. Lenovo ID is 0x0848. Please post a screenshot of Memtest86+ running on one of these module to grab the correct ID.

Here is a picture of this. Idk if it's a correct JEDEC ID or not, but I know Lenovo definitely uses that, as I have one Lenovo right in front of me with such memory.

![Image](https://github.com/user-attachments/assets/666c32cc-25f2-4fbc-b883-1d54151ced19)

Here's also a picture of the RAM module itself.

![Image](https://github.com/user-attachments/assets/6f114378-5336-461a-bd99-26d42796f85d)

**Comment by dmsc** [comment]
Hi!

I have a PUSKILL SO-DIMM module, ID is 0xA2D, can you un-comment it from the list?

<img width="1289" height="611" alt="Image" src="https://github.com/user-attachments/assets/40ba9a94-33af-4204-b89b-fcf81c72e3dd" />

![Image](https://github.com/user-attachments/assets/8d81c8b9-97ec-46b8-ad58-d0f1d74e19ca)

Thank you!

**Comment by x86fr** [comment]
Noted.

**Comment by shengvictory-bit** [comment]
Please add the ID code

    { 0x0D61, "Lyczar" },

Thank you!

**Comment by x86fr** [comment]
NTS: Add 0x0A5D (SZ DOKE samples)

**Comment by daiaji** [comment]
I would like to request the re-enabling (uncommenting) of several JEDEC Manufacturer IDs in `system/jedec_id.h`. Currently, many high-volume, globally active manufacturers are commented out, leading to frequent "Unknown" reports in Memtest86+ v8.00. 

While keeping the binary small is important, the current whitelist is inconsistent (e.g., **Kingbank 0x0B12** and **SOYO 0x0B58** are enabled, while the following larger players are disabled).

#### **1. Major Retail & Gaming Brands (Active)**
*   **`0x0A7D` - Juhor (玖合) / Seiwhale (枭鲸)**: Massive presence in budget/performance markets. My module (ID `0xa7d`) is currently reported as Unknown.
*   **`0x0871` - Asgard (阿斯加特)**: A leading gaming brand globally (Gloway group).
*   **`0x083C` / `0x083D` - Cuso/Kuso (酷兽)**: High-volume budget modules.
*   **`0x0A31` - Biwin (佰维存储)**: **Critical.** They manufacture and brand memory for **Acer, Predator, and HP**.
*   **`0x0A5D` - SKIHOTAR (天策/思瀚微)**: Increasingly common in online retail (AliExpress/JD).

#### **2. Motherboard & OEM Vendors (Active)**
*   **`0x096C` - Colorful (七彩虹)**: Major motherboard/GPU vendor with significant RAM output.
*   **`0x0922` - Maxsun (铭瑄)**: Very active in the DIY PC market.
*   **`0x0B2C` - Hikstorage (海康存储)**: Global storage giant (Hikvision group).
*   **`0x0B77` - Dahua (大华存储)**: Significant industrial and retail presence.
*   **`0x0C7D` - Dell (戴尔)**: Many Dell OEM modules use this ID. Commenting it out is highly problematic for Dell systems.

#### **3. International/OEM Specialists (Active)**
*   **`0x0132` - Mushkin Enhanced Memory**: Established enthusiast brand.
*   **`0x0577` - Avant**: One of the world's largest OEM memory manufacturers.

---

#### **Verification for ID `0x0A7D` (Juhor/Seiwhale):**
The following `dmidecode` output confirms the JEDEC ID for my specific module:

```text
Handle 0x001A, DMI type 17
        Manufacturer: Unknown
        Part Number: SED3200S2216        
        Module Manufacturer ID: Bank 11, Hex 0xFD (Parsed as 0x7D / 0xa7d)
```

**Proposed Action:**
Please uncomment the lines for **0x0A7D, 0x0871, 0x083C, 0x083D, 0x0A31, 0x096C, 0x0922, 0x0B2C, 0x0B77, 0x0132, 0x0577,** and **0x0C7D**. This would significantly improve hardware detection accuracy for a vast number of modern users.

**Supporting Images:**

![Image](https://github.com/user-attachments/assets/72e7183e-57f8-4dca-a353-b33a015fbb0e)

![Image](https://github.com/user-attachments/assets/f0206fb4-53b9-4de3-9472-4f0156af4f38)

**Comment by x86fr** [comment]
Thank you for that list. All of them (and a few others have been enabled for v8.10)

**Comment by x86fr** [comment]
----------- V8.10 Released ------------------

NTS: Add 0x081A
